GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m

```diff
@@ -99,28 +100,24 @@ + (instancetype)sharedInstance {
   return sharedInstance;
 }
 
-- (instancetype)init {
-  self = [super init];
-  if (self) {
-    _uploaderQueue = dispatch_queue_create("com.google.GDTCCTUploader", DISPATCH_QUEUE_SERIAL);
-  }
-  return self;
++ (void)setTestServerURL:(NSURL *_Nullable)serverURL {
+  _testServerURL = serverURL;
 }
 
-- (NSURLSession *)uploaderSession {
-  if (_uploaderSession == nil) {
-    NSURLSessionConfiguration *config = [NSURLSessionConfiguration ephemeralSessionConfiguration];
-    _uploaderSession = [NSURLSession sessionWithConfiguration:config
-                                                     delegate:self
-                                                delegateQueue:nil];
-  }
-  return _uploaderSession;
++ (NSURL *_Nullable)testServerURL {
+  return _testServerURL;
 }
 
-/**
- *
- */
-- (nullable NSURL *)serverURLForTarget:(GDTCORTarget)target {
++ (NSDictionary<NSNumber *, NSURL *> *)uploadURLs {
```

Comment:
I like this idea of having a dictionary. The only concern is replication. If we add a new target and endpoint, we might have to remember to update the endpoint and target mapping in two separate places which might not be the most ideal one.

Can we make one reference the other and have a single source of truth?

Reply:
This is a good point! I think we can restructure the code so that `serverURLForTarget` uses `uploadURLs` dictionary under the hood. WDYT?

Reply:
I like that idea. @jeremyjiang-dev WDYT?

Reply:
Thanks Visu and Maksym for the great idea! I have updated the changes.